src/gar.lua

```diff
@@ -1109,6 +1110,23 @@ function gar.allowDelegates(delegates, gatewayAddress)
 	}
 end
 
+function gar.isEligibleForArNSDiscount(from)
+	local gateway = gar.getGateway(from)
+	if gateway == nil then
+		return false
+	end
+
+	if gateway.weights == nil then
+		return false
+	end
```

Comment:
nit, can combine to the same if
```suggestion
	if gateway == nil or gateay.weights == nil then
		return false
	end
```